File: app/src/main/java/com/qw/ActivityMain.java
package com.qw;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;


import com.qw.tools.R;
import com.test.soultools.tool.log.TLog;


/**
 * Author: 思忆
 * Date: Created in 2020/5/14 4:24 PM
 */
public class ActivityMain extends AppCompatActivity {
    Button buttonTest;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        findViewById(R.id.btn2).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(ActivityMain.this, ActivitySecond.class));
            }
        });
        buttonTest = findViewById(R.id.btn1);
        buttonTest.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                TLog.d("qw", "This is An Log in java:",ActivityMain.this);
                Toast.makeText(ActivityMain.this, "", Toast.LENGTH_SHORT).show();
            }
        });


    }


}

File: app/src/main/java/com/qw/ActivityMain.kt
package com.qw

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.qw.tools.R
import com.test.soultools.tool.log.TLog
import com.test.soultools.tool.log.TLog.d
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext

/**
 * Author: 思忆
 * Date: Created in 2020/5/14 4:24 PM
 */
class ActivityMain : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        findViewById<View>(R.id.btn2).setOnClickListener { startActivity(Intent(this@ActivityMain, ActivitySecond::class.java)) }
        findViewById<View>(R.id.btn1)?.setOnClickListener {
            startActivity(Intent(this@ActivityMain, ActivityThree::class.java))
        }

    }


}